✨ Feat: #102 RecordDetailViewModel 개선 및 MyRecordView 기능 구현

Mark3891 opened the pull request:
<!--
🙏 PR 제목 컨벤션 (Gitmoji + 타입 + 이슈 번호 + 작업 요약)
예시: ✨ Feat: #167 예약 취소 구현
※ PR 생성 시 Assignees 및 Labels 설정도 잊지 마세요!
-->

## ✨ What’s this PR?
### 📌 관련 이슈 (Related Issue)
<!-- 해당 PR이 어떤 이슈를 해결하는지 연결해주세요 -->
- Closes #102 

---

### 🧶 주요 변경 내용 (Summary)
<!-- 이번 PR에서 작업한 핵심 변경 사항을 작성해주세요 -->

- RecordDetailViewModel의 초기화 인자를 id 하나로 단순화하여 구조 개선

- MyRecordView에 Mock 데이터를 연결하고 최신순 정렬 적용

- MyRecordCard 클릭 시 RecordDetailBottomSheet가 올라오도록 UI 연동 완료

---

### 📸 스크린샷 (Optional)
<!-- UI 작업의 경우, 구현한 화면을 첨부해주세요 -->
<!-- 이미지 크기 조절 예시: <img width="300" alt="설명" src="링크"> -->
<img width="300" alt="image" src="https://github.com/user-attachments/assets/3017655a-ca7f-4b02-9d35-d0cbb21de53e" />

---

### 🧪 테스트 / 검증 내역
<!-- 동작 확인 여부나 시나리오 테스트 내용을 간단히 써주세요 -->

- [x] UI 정상 동작 확인
- [x] iPhone 15, iOS 17.4 환경에서 정상 동작
- [ ] 다크모드 테스트는 이후 이슈로 분리 예정 (#000)

---

### 💬 기타 공유 사항
<!-- 리뷰어가 참고하면 좋을 정보, 고민했던 지점 등을 적어주세요 -->

- MyRecordCard 내부에서 onTapGesture로 selectedRecordID 바인딩하여 BottomSheet 띄움

- 현재는 Mock 데이터로 동작하며, 추후 API 연동 시 별도 처리 필요


---

### 🙇🏻‍♀️ 리뷰 가이드 (선택)
<!-- 리뷰어가 중점적으로 보면 좋을 포인트가 있다면 알려주세요 -->

Mark3891 committed:
✨ Feat. RecordDetailViewModel 개선 및 MyRecordView 기능 추가

✨ Feat. RecordDetailViewModel 개선 및 MyRecordView 기능 추가

Why: RecordDetailViewModel에서 id만으로 데이터 로딩이 가능하여 인터페이스를 간소화함.
또한, MyRecordView에 목데이터를 연결하고 최신순 정렬로 UX를 개선하고자 함.

How:
- RecordDetailViewModel에서 id만 인자로 받도록 수정
- MyRecordView에 MockData 연결
- 데이터 최신순으로 정렬 처리
- MyRecordCard 탭 시 RecordDetailBottomSheet 노출되도록 연동

Tag: #swift #feat #viewmodel #mockdata #ui

freeskyES commented on Packages/Features/Sources/Features/Map/MapViewRepresentable.swift:
```diff
@@ -4,32 +4,24 @@ import MapKit
         return mapView
     }
     
     func updateUIView(_ uiView: MKMapView, context: Context) {
```
이 파일 안에서도 이미 다양한 로직을 함수로 잘 나눠주셨지만,
일부 메서드에서는 여전히 여러 역할이 함께 수행되고 있어서,
각 책임을 명확히 분리된 private 메서드로 분리해주시면 가독성과 테스트 용이성이 더 높아질 것 같습니다

예시)
```swift
func updateUIView(_ uiView: MKMapView, context: Context) {
    updateAnnotationsIfNeeded(on: uiView)
    moveToUserLocationIfNeeded(on: uiView)
    moveToCameraPositionIfNeeded(on: uiView)
}
```
각 역할을 의미 있는 이름의 메서드로 분리하면 코드 흐름이 더 명확하게 전달될 수 있을 것 같아요!

freeskyES commented on Packages/Features/Sources/Features/Map/MapViewRepresentable.swift:
```diff
@@ -38,79 +30,65 @@ struct MapViewRepresentable: UIViewRepresentable {
     
-    // MARK: - Helper
-    
     private func updateAnnotations(from mapView: MKMapView) {
```
예시 2)
```swift
private func updateAnnotations(from mapView: MKMapView) {
    let existing = getExistingAnnotations(from: mapView)
    let (toAdd, toRemove) = calculateAnnotationDifferences(existing: existing)
    applyAnnotationChanges(to: mapView, add: toAdd, remove: toRemove)
}
```

freeskyES commented on Packages/Features/Sources/Features/Map/MapViewRepresentable.swift:
```diff
@@ -120,32 +98,86 @@ struct MapViewRepresentable: UIViewRepresentable {
+                annotationView?.clusteringIdentifier = "customPinCluster"
+                annotationView?.frame.size = CGSize(width: 40, height: 40)
+                
+                let titleLabel = UILabel()
```
예시 3)
```swift
private func makeTitleLabel() -> UILabel {
    let label = UILabel()
    label.tag = 1001
    label.font = UIFont.systemFont(ofSize: 12)
    label.textColor = .black
    label.textAlignment = .center
    label.backgroundColor = UIColor.white.withAlphaComponent(0.0)
    label.layer.cornerRadius = 4
    label.layer.masksToBounds = true
    label.frame = CGRect(x: -30, y: 60, width: 100, height: 20)
    return label
}
```

Mark3891 committed:
♻️ Refactor. MapViewRepresentable 역할별 함수 분리

♻️ Refactor. MapViewRepresentable 역할별 함수 분리

Why: updateUIView 및 Annotation 관련 코드에서 하나의 함수가 여러 역할을 수행하고 있어
     가독성과 유지보수성이 떨어졌음

How:
- updateUIView를 updateAnnotationsIfNeeded / moveToUserLocationIfNeeded / moveToCameraPositionIfNeeded로 분리
- Annotation 업데이트 로직을 getExistingAnnotations, calculateAnnotationDifferences, applyAnnotationChanges로 분리
- Coordinator에서 Cluster 및 Custom Pin 뷰 구성도 전용 메서드로 모듈화
- UILabel 생성 로직을 makeTitleLabel()로 분리

Tag: #refactor #mapkit #swiftui #coordinator

Mark3891 committed:
♻️ Refactor. MapViewRepresentable 역할별 함수 분리 및 불변 변수 최적화

♻️ Refactor. MapViewRepresentable 역할별 함수 분리 및 불변 변수 최적화

Why:
- updateUIView 및 Annotation 관련 함수들이 한 함수에서 여러 역할을 처리하고 있어 가독성이 떨어졌음
- 불변 변수임에도 var로 선언된 부분에서 Swift 컴파일러 경고 발생

How:
- updateUIView를 updateAnnotationsIfNeeded / moveToUserLocationIfNeeded / moveToCameraPositionIfNeeded로 분리하여 책임 분산
- Annotation 관련 로직을 getExistingAnnotations, calculateAnnotationDifferences, applyAnnotationChanges로 분리
- Cluster 및 CustomPin 뷰 생성을 전용 함수로 분리: createClusterView, createCustomPinView, makeTitleLabel
- 불변 변수인 clusterView와 annotationView를 let으로 선언하여 컴파일러 경고 제거

Tag: #refactor #swift #mapkit #annotation #clean-code
